### ISSUE TITLE
Emit traffic events on generated api

### ISSUE THREAD

**Issue by miralemd** [qlik-oss/enigma.js#537]
I would like to be able to hook into the session traffic for any of the generated APIs, currently only `changed` and `closed` are emitted.

```js
api.on('traffic:received', (data) => {
  // do magic
});
```

**Comment by axelssonHakan** [comment]
From README.md
```
// bind traffic events to log what is sent and received on the socket:
session.on('traffic:sent', data => console.log('sent:', data));
session.on('traffic:received', data => console.log('received:', data));
```

Is this on "wrong level"?

**Comment by peol** [comment]
@axelssonHakan Yeah, me and @miralemd discussed a `api.on('traffic', fn)` event to avoid having to check the session traffic events, `traffic:sent` for the `handle` and `id`, and `traffic:received` for verifying the response by `id` and simply do `api.on('traffic:received, fn)` to get all (sent) traffic to that specific API.
